File: Domain/Tooling.cs
namespace Domain
{
    public class Tooling
    {
        public Guid Id { get; set; }
        public string TNumber { get; set; }
        public string PSNumber { get; set; }
        public int Quantity { get; set; }
        public string Department { get; set; }
        public string Note { get; set; }
        public bool IsInProduction { get; set; }
        public int NumberOfImpressions { get; set; }
        public string PunnetNumber { get; set; }
        public ICollection<Image> Images { get; set; } = new List<Image>();
        public ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: client-app/src/models/tooling.ts
export interface Tooling {
  id?: string | undefined;
  tNumber: string;
  psNumber: string;
  quantity: number | undefined;
  department: string;
  note?: string;
  isInProduction: boolean;
  numberOfImpressions: number | undefined;
  image?: string;
  punnetNumber: string;
  products?: Product[];
  images?: Image[];
}

export interface Product {
  id?: number;
  name: string;
  isAllergen: boolean;
  toolingId?: string | undefined;
}


export interface Image {
  id: string;
  url: string;
  isMain: boolean;
}

export interface ImageUpload {
  files: Blob[];
  id: string | undefined;
}
